src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,149 @@
+ */
+public class Parser {
+
+    /**
+     * Extracts the command specified in the user input and creates the respective command objects.
+     * @param input user input from command line
+     * @return respective Command class objects
+     */
+    public static Command parse(String input) {
+        try {
+            String[] taskInfo = input.split(" ", 2);
+            if (taskInfo[0].equals("exit")) {
+                return new ExitCommand();
+            } else if (taskInfo[0].equals("help")) {
+                // CREATE A HELP COMMAND TO SHOW THE AVAILABLE INSTRUCTION
+                return null;
+            } else if (taskInfo[0].equals("add")) {
+                if (taskInfo.length == 1) {
+                    throw new WrongAddFormatException();
+                }
+                String[] wordDetail = taskInfo[1].split("w/");
+                if (wordDetail.length != 3) {
+                    throw new WrongAddFormatException();
+                }
+                String wordDescription = wordDetail[1].trim();
+                if (wordDescription.length() == 0) {
+                    throw new EmptyWordException();
+                }
+                String[] meaningAndTag = wordDetail[2].split("t/");
+                String meaning = meaningAndTag[0].trim();
+                if (meaning.length() == 0) {
+                    throw new EmptyWordException();
+                }
+                Word word;
+                if (meaningAndTag.length > 1) {
+                    HashSet<String> tags = new HashSet<>();
+                    for (int j = 1; j < meaningAndTag.length; ++j) {
+                        tags.add(meaningAndTag[j]);
+                    }
+                    word = new Word(wordDescription, meaning, tags);
+                }
+                else {
+                    word = new Word(wordDescription, meaning);
+                }
+                return new AddCommand(word);
+            } else if (taskInfo[0].equals("delete")) {
+                if (taskInfo.length == 1 || !taskInfo[1].startsWith("w/")) {
+                    throw new WrongDeleteFormatException();
+                }
+                String[] wordAndTags = taskInfo[1].split("t/");
+                //delete word
+                if (wordAndTags.length == 1) {
+                    return new DeleteCommand(taskInfo[1].substring(2));
+                }
+                //delete tag
+                else {
+                    String wordDescription = wordAndTags[0].substring(2).trim();
+                    ArrayList<String> tags = new ArrayList<>();
+                    for (int i = 1; i < wordAndTags.length; ++i) {
+                        tags.add(wordAndTags[i].trim());
+                    }
+                    return new DeleteCommand(wordDescription, tags);
+                }
+            } else if (taskInfo[0].equals("search")) {
+                if (taskInfo.length == 1 || !taskInfo[1].startsWith("w/")) {
```

Comment:
Because the each case has like 10+ lines of code, consider putting them in separate private methods